### PR TITLE
[Infra-719] pg10 metrics

### PR DESCRIPTION
**Pull request by gesanderson**
* Make postgres collectors compatible with postgres 10
* Add new metrics for connection's wait event type

### CODE REVIEW

**Comment by gesanderson** [comment]
@mathpl @jpdionne waiting for review on this

File: collectors/0/postgresql_replication.py [hunk 1]
```diff
@@ -44,10 +44,16 @@ def collect(db):
     cursor = db.cursor()
 
     # Replication lag time (could be slave only or a master / slave combo)
-    cursor.execute("SELECT "
-                   "CASE WHEN pg_is_in_recovery() THEN (EXTRACT (EPOCH FROM now() - pg_last_xact_replay_timestamp()) * 1000)::BIGINT ELSE NULL END AS replication_lag_time, "
-                   "pg_xlog_location_diff(pg_last_xlog_receive_location(), pg_last_xlog_replay_location()) AS replication_lag_bytes, "
-                   "pg_is_in_recovery() AS in_recovery;")
+    if is_pg10(cursor):
```
Comment (jpdionne) [inline collectors/0/postgresql_replication.py]: Could you determine the version only once after db = connect()?

Comment (gesanderson) [reply collectors/0/postgresql_replication.py]: I could, but to me this seems like a micro-optimization

Comment (jpdionne) [reply collectors/0/postgresql_replication.py]: We log all those queries, that is why I was proposing and we know the version won't change at each check cycles.

**Pull request merged**